### DISCUSSION TITLE
How can GitHub Actions be used to automate the process of testing and deploying an application when a new commit is made to the main branch?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#143008] [New to GitHub]
How can GitHub Actions be configured for a Flutter project to automate the processes of testing, building, and deploying the application when a new commit is pushed to the main branch? Specifically, what steps need to be taken for GitHub Actions to run automated tests (unit tests, widget tests), build the application for both Android and iOS platforms, and then deploy the application to environments like Firebase App Distribution or Google Play once all tests and builds have successfully completed?

**Comment by datonefeed** [comment]
To set up GitHub Actions for a Flutter project to automate testing, building, and deployment:

Create Workflow File: Create a .yml file in .github/workflows/, specifying steps to run tests, build, and deploy on push to the main branch.
Testing: Use flutter test to run unit and widget tests.
Build: Build Android APK (flutter build apk --release) and iOS app (flutter build ios --release --no-codesign).
Deploy: Use Firebase App Distribution or integrate with platforms like Google Play or the App Store.
Secrets: Add necessary secrets (e.g., Firebase tokens) in GitHub repository settings for secure deployment.
This creates a full CI/CD pipeline for your Flutter project.

**Reply by ghost** [reply]
thanks

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/143007#discussioncomment-11092044) for reference to why this discussion was closed and locked. Thanks!
